File: bot/src/main/java/by/yayauheny/tutochkatgbot/dto/backend/NearestRestroomSlimDto.java
package by.yayauheny.tutochkatgbot.dto.backend;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.UUID;

/**
 * Slim DTO for nearest restrooms list.
 * Contains only fields necessary for list UI and caching.
 */
public record NearestRestroomSlimDto(
    @JsonProperty("id") UUID id,
    @JsonProperty("displayName") String displayName,
    @JsonProperty("distanceMeters") Double distanceMeters,
    @JsonProperty("feeType") FeeType feeType,
    @JsonProperty("coordinates") LatLon coordinates,
    @JsonProperty("subwayStation") SubwayStationSlimDto subwayStation
) {}

File: backend/src/main/kotlin/yayauheny/by/model/dto/NearestRestroomSlimDto.kt
package yayauheny.by.model.dto

import java.util.UUID
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import yayauheny.by.model.enums.FeeType

/**
 * Slim DTO for nearest restrooms list.
 * Contains only fields necessary for list UI and caching.
 */
@Serializable
data class NearestRestroomSlimDto(
    @Contextual val id: UUID,
    /**
     * Restroom name from DB.
     */
    val displayName: String,
    val distanceMeters: Double,
    val feeType: FeeType,
    val queryCoordinates: Coordinates,
    val restroomCoordinates: Coordinates
)
